server/queue_priv.php

```diff
@@ -1,13 +1,83 @@
 <?php
```

Comment:
You should check to make sure a user is logged in before letting them continue on with calling this class.
Check out management/appointments/index.php. That does a permission check at the top, but you should do instead
`$USER->isLoggedIn()` and if the user is not, then don't let them call this page.